components/nearby-diagram.js

```diff
@@ -319,6 +319,40 @@ export default function NearbyDiagram({
   const viewBoxHeight =
     motifPositionY + motifsList.length * geneUnitHeight + blankHeight * 2;
 
+  function getGeneScaleLabel(
+    index,
+    tickWidth,
+    scaleForGene,
+    geneScaleStartTick
+  ) {
+    const num = Math.floor(
+      (index * tickWidth) / scaleForGene + geneScaleStartTick
+    );
+    return num >= 0 ? num : "";
+  }
+
+  function getVariantScaleLabel(
+    index,
+    tickWidth,
+    scaleForVariantInLd,
+    offsetForVariantInLd
+  ) {
+    const num = Math.floor(
+      (index * tickWidth) / scaleForVariantInLd + offsetForVariantInLd
+    );
+    return num >= 0 ? num : "";
+  }
+
+  function getSequenceScaleLabel(
+    index,
+    tickWidth,
+    baseWidth,
+    offsetXForVariant
+  ) {
+    const num = Math.floor((index * tickWidth) / baseWidth + offsetXForVariant);
+    return num >= 0 ? num : "";
+  }
+
```

Comment:
Just noting these functions return a number if valid, and a string if invalid, and I see why this is convenient from your usage. But built-in Javascript functions that return numbers only return numbers, with usually -1 or some other special value if invalid (e.g. `findIndex()`). This also makes Typescript conversions much easier. I realize this means you probably have to assign their return values to a variable, and then conditionally render based on the special invalid value — not as convenient, so that’s why I’m just noting this.